docs/docs/reference/production-deployment.md

```diff
@@ -0,0 +1,97 @@
+
+### Proxy
+
+Proxy service, as the name implies, is responsible for proxying all user traffic, in addition to performing Authentication and Authorization checks.  The proxy is directly in path for user traffic.
+
+Proxy will need resources scaled in conjunction with request count and may need average request size accounted for.  The heavier your user traffic, the more resources the Proxy service should have provisioned.
+
+### Authorize
+
+The Authorize serivce is responsible for policy checks during requests.  It is in the hot path for user requests but does not directly handle user traffic.
+
+Authorize will need resources scaled in conjunction with request count.  Request size and type should be of a constant complexity.  Policy complexity/length may have some impact on performance.
+
+### Authenticate
+
+The Authenticate service handles session token renewal, creation, and interaction with your IDP.  The Authenticate service between a successful session setup or renewal and session expiration.
+
+Authenticate requires significantly less resources than other components due to the only-occasional requirement to establish/renew sessions.  Add resources to the Authenticate service if you have a high session churn rate.  The requests should be constant time and complexity, but may vary by IDP implementation.
+
+## Load Balancing
+
+### Proxy
+
+You should provide a TCP or HTTP(s) load balancer between end users and the Proxy service.
+
+Proxy can handle its own SSL termination but is not a full web server.  If you need any special capabilities such as redirects, sticky sessions, etc, it is typical to put Pomerium behind an L7 load balancer or ingress controller.
+
+
+### Authenticate
+
+You should provide a TCP or HTTP(s) load balancer between end users and the Authorize service.
+
+Authenticate is compatible with any L4 or L7/HTTP load balancer.  Session stickiness should not be required and it is typical to have Authenticate be a named vhost on the same L7 load balancer as the Proxy service.
+
+### Authorize
+
+You do **not** need to provide a load balancer between the Proxy and the Authorize service.  Authorize utilizes GRPC and thus has special requirements if you should chose to use an external load balancer.  GRPC can perform client based load balancing, and in most configurations is the best architecture.
+
+By default, the Proxy will automatically connect to all IPs returned by a DNS query for the name of the Authorize serivce.  Proxy will then regularly re-query DNS for changes to the Authorize service cluster.  Health checks and failover are automatic.
+
+If you chose to use an external proxying load balancer instead of the default client implementation, it must be able to understand HTTP2 to properly distribute requests.  In L4 load balancing configurations, GRPC/HTTP2 traffic will be pinned to a single Authorize instance, due to the way HTTP2 multiplexes requests over a single established connection.  
+
+## High Availability
```

Comment:
Whole section is well said. Nice work. 👍

Reply:
Made two tweaks

Reply:
LGTM